src/main/java/eu/darkbot/api/managers/AttackAPI.java

```diff
@@ -2,105 +2,87 @@
 
 import eu.darkbot.api.API;
 import eu.darkbot.api.entities.Entity;
+import eu.darkbot.api.entities.utils.Ammo;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
-// TODO: 07.11.2020 better name sadge
-public interface AttackAPI extends API {
-
-    @Nullable
-    Entity getTarget();
-    boolean setTarget(@Nullable Entity entity);
-
-    boolean isTargeted();
-    boolean clickTarget();
-
-    boolean isAttacking();
-    boolean laserAttack();
-    boolean laserAbort();
+import java.util.Optional;
 
-    boolean canBeClicked(@NotNull LaserAmmo ammo); //is given ammo in slotbar and can be handled via keyboard.
-    LaserAmmo setLaserAmmo(@NotNull String laserId);
-    LaserAmmo setLaserAmmo(@NotNull AttackAPI.LaserAmmo laserAmmo);
-
-    boolean canBeClicked(@NotNull RocketAmmo ammo);
-    RocketAmmo setRocketAmmo(@NotNull String rocketId);
-    RocketAmmo setRocketAmmo(@NotNull AttackAPI.RocketAmmo rocketAmmo);
-
-    enum LaserAmmo {
-        UNKNOWN,
-        LCB_10,
-        MCB_25,
-        MCB_50,
-        UCB_100,
-        SAB_50,
-        CBO_100,
-        RSB_75(true),
-        JOB_100,
-        RB_214,
-        PIB_100,
-        SPACECUP_GROUP_A,
-        SPACECUP_GROUP_B,
-        SPACECUP_GROUP_C,
-        SPACECUP_GROUP_D,
-        SPACECUP_GROUP_E,
-        SPACECUP_GROUP_F,
-        SPACECUP_GROUP_G,
-        SPACECUP_GROUP_H,
-        A_BL,
-        RCB_140,
-        IDB_125,
-        VB_142,
-        EMAA_20,
-        SBL_100;
+public interface AttackAPI extends API {
 
-        private final boolean cooldown;
+    boolean hasTarget();
 
-        LaserAmmo() {
-            this(false);
-        }
+    /**
+     * @return currently set target
+     */
+    Optional<Entity> getTarget();
 
-        LaserAmmo(boolean cooldown) {
-            this.cooldown = cooldown;
-        }
+    /**
+     * @param entity to be set
+     * @return previously set {@link Entity}
+     */
+    Optional<Entity> setTarget(@Nullable Entity entity);
 
-        public String getId() {
-            return "ammunition_laser_" + name().toLowerCase().replace("_", "-");
-        }
+    /**
+     * @return true if selected {@link Entity} is clicked/targeted
+     */
+    boolean isTargeted();
 
-        public boolean hasCooldown() {
-            return cooldown;
-        }
-    }
+    /**
+     * Tries to click selected target
+     *
+     * @return true if click was successful
+     * @see #getTarget()
+     */
+    boolean clickTarget();
```

Comment:
Like suggested before, don't name this "click", there may not be a mouse, use `target` or `attack`